util/all_tests.json

```diff
@@ -72,25 +72,13 @@
     "skip_valgrind": true
   },
   {
-    "comment": "Run RAND test suite with only madv WIPEONFORK enabled",
-    "cmd": ["crypto/crypto_test", "--gtest_filter=RandTest.*"],
-    "env": ["BORINGSSL_IGNORE_PTHREAD_ATFORK=1"],
-    "skip_valgrind": true
-  },
-  {
-    "comment": "Run RAND test suite with only pthread_atfork enabled",
+    "comment": "Run RAND test suite without MADV_WIPEONFORK enabled",
     "cmd": ["crypto/crypto_test", "--gtest_filter=RandTest.*"],
     "env": ["BORINGSSL_IGNORE_MADV_WIPEONFORK=1"],
     "skip_valgrind": true
   },
   {
-    "comment": "Run fork detection test suite with only madv WIPEONFORK enabled",
-    "cmd": ["crypto/crypto_test", "--gtest_filter=ForkDetect.*"],
-    "env": ["BORINGSSL_IGNORE_PTHREAD_ATFORK=1"],
-    "skip_valgrind": true
-  },
-  {
-    "comment": "Run fork detection test suite with only pthread_atfork enabled",
+    "comment": "Run fork detection test suite without MADV_WIPEONFORK enabled",
     "cmd": ["crypto/crypto_test", "--gtest_filter=ForkDetect.*"],
     "env": ["BORINGSSL_IGNORE_MADV_WIPEONFORK=1"],
     "skip_valgrind": true
```

Comment:
What is is this test doing if the only fork detection logic is disabled?

Reply:
It tests the fall-back logic to prediction resistance. That is, the tests performs some forking and other uniqueness breaking events that could cause state duplication. We must still protect against this even though fork detection mechanism are not supported.

Right now, this is somewhat indistinguishable from the test case with `--fork_unsafe_buffering`. But I touched a bit on this in the PR description:
> Remove pthread_atfork() specific tests and replace with tests that forcefully disables MADV_WIPEONFORK to test those code-paths (which in the future could potentially be different from --fork_unsafe_buffering).